salt/client/mixins.py

```diff
@@ -359,29 +359,20 @@ def _low(self, fun, low, print_event=True, full_return=False):
             # packed into the top level object. The plan is to move away from
             # that since the caller knows what is an arg vs a kwarg, but while
             # we make the transition we will load "kwargs" using format_call if
-            # there are no kwargs in the low object passed in
-            f_call = None
-            if 'arg' not in low:
-                f_call = salt.utils.format_call(
+            # there are no kwargs in the low object passed in.
+            f_call = {} if 'arg' in low and 'kwarg' in low \
+                else salt.utils.format_call(
                     self.functions[fun],
                     low,
                     expected_extra_kws=CLIENT_INTERNAL_KEYWORDS
                 )
-                args = f_call.get('args', ())
-            else:
-                args = low['arg']
-
-            if 'kwarg' not in low:
-                log.critical(
-                    'kwargs must be passed inside the low data within the '
-                    '\'kwarg\' key. See usage of '
-                    'salt.utils.args.parse_input() and '
-                    'salt.minion.load_args_and_kwargs() elsewhere in the '
-                    'codebase.'
-                )
-                kwargs = {}
-            else:
-                kwargs = low['kwarg']
+
+            args = f_call.get('args', ()) \
+                if 'arg' not in low \
+                else low['arg']
+            kwargs = f_call.get('kwargs', {}) \
+                if 'kwarg' not in low \
+                else low['kwarg']
```

Comment:
I like the logic. Here is a simplification that I can follow a bit easier and should reduce the lines of code:

```
if 'arg' in low and 'kwarg' in low:
    args = low['arg']
    kwargs = low['kwarg']
else:
    f_call = salt.utils.format_call(
        self.functions[fun],
        low,
        expected_extra_kws=CLIENT_INTERNAL_KEYWORDS
    )
    args = f_call.get('args', ())
    kwargs = f_call.get('kwargs', {})
```

Reply:
Good call, I've pushed a commit that makes this change.